src/dos/cdrom_image.cpp

```diff
@@ -1064,17 +1202,14 @@ bool CDROM_Interface_Image::GetCueString(string &str, istream &in)
 
 void CDROM_Interface_Image::ClearTracks()
 {
-	vector<Track>::iterator i = tracks.begin();
-	vector<Track>::iterator end = tracks.end();
-
-	TrackFile* last = nullptr;
-	while(i != end) {
-		Track &curr = *i;
-		if (curr.file != last) {
-			delete curr.file;
-			last = curr.file;
+	track_iter track = tracks.begin();
+	track_const_iter end = tracks.end();
+	while(track != end) {
```

Comment:
Like in the previous comment, using foreach loop will make this code shorter and a bit nicer :)

Reply:
very nice! I haven't used this syntax, so need to get comfortable with it.
The "auto" everything approach feels like the wheels have flown off; but I just have to trust the compiler to work out the object's iterator types.
Fixed in 7d257ac5